### ISSUE TITLE
Large entity tables cause action buttons to stack unnecessarily

### ISSUE THREAD

**Issue by geraldhumphries** [jhipster/generator-jhipster#2267]
##### Overview of the issue
This is a minor issue. If the table in `entities.html` is large enough to scroll horizontally, the View, Edit, and Delete buttons at the end of the table will stack on top of each other instead of remaining beside each other.

##### Motivation for or Use Case
This makes the rows a lot taller which looks ugly and affects usability.

![stacked-buttons](https://cloud.githubusercontent.com/assets/5278953/10889998/40c2c1ba-8163-11e5-8bd6-0b24b6865e8e.png)

##### JHipster Version
2.23.0

##### Reproduce the error
Create an entity with enough fields to make the table scroll horizontally, then add a row and visit the entity list

##### Suggest a Fix
this can be solved with `white-space: nowrap` on the td. Probably requires a new class on the table or td.

```
table.entity td:last-child {
    white-space: nowrap;
}

<table class="table table-striped entity">...
```

or 

```
td.action-buttons {
    white-space: nowrap;
}
<td class="action-buttons">...
```

![unstacked-buttons](https://cloud.githubusercontent.com/assets/5278953/10890005/4fa07524-8163-11e5-9017-1b630b0ba7cd.png)

**Comment by mraible** [comment]
+1. I also think these could have a couple additional improvements:

1. Make the button's smaller by adding the "btn-sm" class.
2. Remove the text and put it in a "tooltip" attribute that will be rendered by UI Bootstrap.

Making these additional changes makes things look better on the desktop, but does make for less touchable area on mobile.

**Comment by deepu105** [comment]
+1 @mraible

**Comment by gzsombor** [comment]
Removing the text will confuse the users - it's far from obvious, what the meaning of the icons for the first time, but otherwise, it's a good idea!

**Comment by geraldhumphries** [comment]
@mraible it looks like the buttons already have `.btn-sm`. They would need to be changed to `.btn-xs`.

![unstacked-buttons-xs](https://cloud.githubusercontent.com/assets/5278953/10921472/bcabfb58-8243-11e5-8b97-f294fa61dd96.png)

Usability isn't too bad on mobile.

**Comment by mraible** [comment]
Yes, `.btn-xs` is what I'm using as well. Here's what my buttons look like (without text). I'm using the [Material Design for Bootstrap theme](https://fezvrasta.github.io/bootstrap-material-design/). 

<img width="317" alt="buttons-without-text" src="https://cloud.githubusercontent.com/assets/17892/10922175/efed0294-8236-11e5-8a5a-c6bfbbb30ffc.png">

**Comment by deepu105** [comment]
I like the way you have stacked search and add, i have done similar in my
project too looks way better than our default. Also +1 for xs btn without
text. Text can be a ui-bootstrap tooltip. I have done that and looks way
cooler

**Comment by deepu105** [comment]
I would also change the delete icon to trash can icon

**Comment by geraldhumphries** [comment]
Maybe we should turn this into a general "improve entity list UI" ticket. @mraible do you think you can open a pull request with your changes?

**Comment by hirro** [comment]
The material design demonstrated by mraible looks so nice, is there any possibility to include that into the generator? 

As it is now, is the title supposed to be on a line separate from the line with search input, search button, and create button? The create button padding has a padding to the left side which looks a bit unbalanced. The search controls are not right aligned, instead they are more centralish aligned.

**Comment by deepu105** [comment]
Its available as a module check the module page

**Comment by hirro** [comment]
Awesome! I was not updated on modules.

**Comment by hirro** [comment]
PR #2570 (fix of issue #2552) also handles this issue.

**Comment by deepu105** [comment]
closing this since this is tracked in #2552 as well
